packages/vgrammar-plot/src/semantic-mark.ts

```diff
@@ -0,0 +1,1282 @@
+              (markSpec.encode.update as any).startAngle = (datum: any, el: IElement, params: any) => {
+                const scale = params[anotherDimScaleId];
+
+                return scale && isDiscrete(scale.type) ? scale.range()[0] + (scale?.bandwidth?.() ?? 0) / 2 : undefined;
+              };
+              (markSpec.encode.update as any).endAngle = (datum: any, el: IElement, params: any) => {
+                const scale = params[anotherDimScaleId];
+
+                return scale && isDiscrete(scale.type) ? scale.range()[1] + (scale?.bandwidth?.() ?? 0) / 2 : undefined;
+              };
+            }
+          }
+          res.push(markSpec);
+        }
+      });
+    }
+
+    return res;
+  }
+
+  protected setDefaultTooltip(): SemanticTooltipOption | Nil {
+    return null;
+  }
+
+  protected parseTooltipSpec(): Array<TooltipSpec | DimensionTooltipSpec> | Nil {
+    const defaultTooltipSpec = this.setDefaultTooltip();
+    const userTooltipSpec = this.spec.tooltip;
+
+    if (userTooltipSpec !== false && userTooltipSpec !== null && defaultTooltipSpec !== null) {
+      const res: Array<TooltipSpec | DimensionTooltipSpec> = [];
+      const tooltipSpec = Object.assign({}, defaultTooltipSpec, userTooltipSpec === true ? {} : userTooltipSpec);
+      const colorChannel = isNil((this.spec.encode as any).color) ? 'group' : 'color';
+      const colorEncode = (this.spec.encode as any)[colorChannel];
+      const dependency = colorEncode ? [this.getScaleId(colorChannel)] : [];
+      const colorAccessor = colorEncode ? getFieldAccessor(colorEncode) : null;
+      const title = {
+        visible: !!tooltipSpec.title || !!tooltipSpec.staticTitle,
+        key: 'title',
+        value: !isNil(tooltipSpec.staticTitle)
+          ? tooltipSpec.staticTitle
+          : {
+              field: (datum: any, el: IElement, params: any) => {
+                return tooltipSpec.title && datum?.length ? getFieldAccessor(tooltipSpec.title)(datum[0]) : undefined;
+              }
+            }
+      };
+      const content =
+        isArray(tooltipSpec.content) && tooltipSpec.content.length
+          ? tooltipSpec.content.map((entry, index) => {
+              return {
+                key: entry.key
+                  ? { field: entry.key }
+                  : !isNil(tooltipSpec.staticContentKey)
+                  ? isArray(tooltipSpec.staticContentKey)
+                    ? tooltipSpec.staticContentKey[index]
+                    : tooltipSpec.staticContentKey
+                  : (datum: any, el: IElement, params: any) => {
+                      return colorAccessor ? colorAccessor(datum) : undefined;
+                    },
+                value: { field: entry.value },
+                symbol: (datum: any, el: IElement, params: any) => {
+                  const scale = params[this.getScaleId(colorChannel)];
+
+                  return {
+                    symbolType: entry.symbol
+                      ? invokeFunctionType(
```

Comment:
This invocation does not seem working, it would directly return `{symbolType: { field: entry.symbol }}`.